### PR TITLE
Misc fixes and tweaks from #6729

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Includes some minor tweaks and fixes from the closed PR #6729

Nothing of substance or that changes behavior. Most relevant changes (and "most" is relative):

- Avoid unnecessary compStore property in gRPC and HTTP API structs
- Remove now-unused KeepWildcardUnescaped property (after #6750)
- Misc improvements to a few unit tests

### CODE REVIEW

File: pkg/http/server.go [hunk 1]
```diff
@@ -257,7 +257,7 @@ func (s *server) useMaxBodySize(r chi.Router) {
 	}
 
 	maxSize := int64(s.config.MaxRequestBodySizeMB) << 20 // To bytes
-	log.Infof("Enabled max body size HTTP middleware with size %d MB", maxSize)
+	log.Infof("Enabled max body size HTTP middleware with size %d bytes", maxSize)
```
Comment (JoshVanL) [inline pkg/http/server.go:260]: Please print the size in MB for logging. Bytes is essentially useless to humans without getting the calculator out at this order of magnitude.

**Review by JoshVanL — DISMISSED** [review]
Small nit from me

File: pkg/http/server.go (hunk 1 above)
Comment (ItalyPaleAle) [reply pkg/http/server.go:260]: Yeah that's a good callout. Feel free to make a suggestion too!
```suggestion
	log.Infof("Enabled max body size HTTP middleware with size %d MB", s.config.MaxRequestBodySizeMB)
```

File: pkg/diagnostics/metrics.go [hunk 2]
```diff
@@ -68,5 +68,11 @@ func InitMetrics(appID, namespace string, rules []config.MetricsRule) error {
 
 	// Set reporting period of views
 	view.SetReportingPeriod(DefaultReportingPeriod)
-	return utils.CreateRulesMap(rules)
```
Comment (shubham1172) [inline pkg/diagnostics/metrics.go:71]: Isn't this change functionally same and more verbose than a direct return statement?

File: pkg/diagnostics/metrics.go [hunk 3]
```diff
@@ -68,5 +68,11 @@ func InitMetrics(appID, namespace string, rules []config.MetricsRule) error {
...
+
+	// Creates a rules map
+	err := utils.CreateRulesMap(rules)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (ItalyPaleAle) [inline pkg/diagnostics/metrics.go:77]:
```suggestion
	return utils.CreateRulesMap(rules)
```

File: pkg/diagnostics/metrics.go (hunk 2 above)
Comment (ItalyPaleAle) [reply pkg/diagnostics/metrics.go]: Reverted it. I do find that a bit clearer to read but not worth the diff.

**Review by shubham1172 — APPROVED** [review]

File: pkg/acl/acl.go [hunk 4]
```diff
@@ -260,7 +260,7 @@ func ApplyAccessControlPolicies(ctx context.Context, operation string, httpVerb
 	var errMessage string
 
 	if err != nil {
-		errMessage = fmt.Sprintf("error in method normalization: %s", err)
+		errMessage = fmt.Sprintf("error in method normalization: %v", err)
```
Comment (DeepanshuA) [inline pkg/acl/acl.go:263]: I know there are couple of `fmt.Sprintf` in our code, and I feel at some places out of those, a string concatenation would have been better probably.
Do we want to do a string concatenation here?

Comment (ItalyPaleAle) [reply pkg/acl/acl.go:263]: Yeah, although for error handling it's probably ok since it should not be hot paths

**Review by DeepanshuA — APPROVED** [review]

File: pkg/http/api_directmessaging.go [hunk 5]
```diff
@@ -41,10 +41,9 @@ func (a *api) constructDirectMessagingEndpoints() []Endpoint {
 			Methods: []string{},
 			Route:   "invoke/*",
 			// This is the fallback route for when no other method is matched by the router
-			IsFallback:            true,
-			Version:               apiVersionV1,
-			KeepWildcardUnescaped: true,
```
Comment (mukundansundar) [inline pkg/http/api_directmessaging.go:46]: is true the only value for this bool in all code paths??

Comment (ItalyPaleAle) [reply pkg/http/api_directmessaging.go:46]: KeepWildcardUnescaped was already a no-op since #6750 and it's "false" for all routes

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**